torch/export/__init__.py

```diff
@@ -155,6 +157,91 @@ def export_for_training(
     )
 
 
+def export_for_inference(
+    mod: torch.nn.Module,
```

Comment:
@tugsbayasgalan I think we can add support for non-nn.Module as well. A lot of people only need to export a free function and if it's ONLY for inference, we can assume they probably don't depend on some node meta that is only available under training.

If you don't have time, we can consider opening a task and leave a TODO.